src/Compiler/Driver/GraphChecking/Graph.fs

```diff
@@ -108,3 +109,16 @@ module internal Graph =
             FileSystem.OpenFileForWriteShim(path, fileMode = System.IO.FileMode.Create)
 
         graph |> serialiseToMermaid |> out.WriteAllText
+
+    let asString (graph: Graph<FileIndex * string>) =
+        let sb = StringBuilder()
+        let appendLine (line: string) = sb.AppendLine(line) |> ignore
+
+        for KeyValue((idx, fileName), _) in graph do
+            appendLine $"%i{idx}[\"%s{fileName}\"]"
+
+        for KeyValue((idx, _), deps) in graph do
```

Comment:
We might print something different here, so far this is partially copypasted from the mermaid serialization.